### ISSUE TITLE
Text copying + UI

### ISSUE THREAD

**Issue by tse-gratis** [AndBible/and-bible#1880]
Thanks so much for the work you are doing, i realize it is a massive effort

Issue #1879 is a consistent problem for me. This issue is a personal niggle, and i realize others may prefer the current situation to this suggestion. Appolgies, i'm going to note a couple of different issues here that are only loosely related 

I would like selecting and copying text to be just that, without the extra popup options and added formatting

The extra formatting, such as beginning and ending "... ..." means i need to re-edit it when pasting it into another document. Also i realize 'Advertise app' is fairly standard and completely your call, but it is fun to imagine leaving the advertisement in all or any of the points throughout my documents :) Aka, I'm not going to do it

My personal choice would be:
* select the text
* menu item contains 'bookmark / compare / copy'
So personally i would not even have a 'Share' option. Though I believe it maybe best left for other users. And also, because i have translator apps installed, i get the extra 'Translate / Translator' options, which to me are pointless here, because im looking at the bible multilingually, so personally again, i would choose to remove those, or just keep the code simple, sticking to the android text selection defaults

Let me quickly note some less related bugs
![photo1634934121 (1)](https://user-images.githubusercontent.com/33900473/138523241-0ce47212-c1dc-4841-9a43-bbeba5f39173.jpeg)
![photo1634934121](https://user-images.githubusercontent.com/33900473/138523249-256e4a3e-a597-4a41-ba30-130c7339f26d.jpeg)

Notice how the verse reference is ```, AraNAV)37-5.34 matta."(```. That should of course be ```matta5.34-37 (AraNAV)```. There are multiple issues with it, such as the numbers getting flipped around. Though I would personally like ```(AraNAV)``` and as much as possible to simply not appear in the copied text. Some other similar issues are that the "... ..." and the first 34 gets flipped to after 35 in the text. I think maybe because the stylized quotes are treated as ltr rather than rtl

The other image only shows very very minor issues, that arent a problem to me. The 'Translate / Translator' -- not a problem. Also notice The verse numberings 29, 31 are on their own line. Also 28 should be down a line and on the hard right, before the text starts. Notice also the ARBWBTC version doesnt have this issue. These things are not a problem, im only mentioning them for completeness, and probably i shouldn't even mention them

---

I hope those things are clear, and since im here, i hope its okay if i make a few related ui suggestions? Feel free to take or ignore them :))

Since the text selection interface has the 'Bookmark / Compare' options, and this pops up when the user does press-and-hold on text, i think it would be great for the user, user interface consistency, and the program code if this was - THE ONLY - way to do bookmarks and verse comparison. Having tap-hold + single tap bookmarks + double tap for full screen + ... makes interaction harder. For instance as a user after multiple tries, and using your app for a long time, i still dont know how to enter and exit full screen without at the same time bringing up the bookmarks dialog

Sorry for merging issues into one feature request, but again, feel free to do whatever you like with this :) I appreciate your work, and have used it extensively with refugees

So on the same concept of streamlining, i would like to suggest the interface elements from the right hand burger menu items, window pane menu items, and bottom right workspaces widget contain crossovers that are removable from a user interface perspective

So, for clarity, that is three user interface items, top (a second hamburger, at same level as hamburger top left), middle (the individual window pane buttons), and bottom (workspace widget)

** top + middle share text-options
Adding confusion is how the top right hamburger text-options only apply to last selected window, whereas middle text-options apply to a specific windows. To note i do actually find text sizes useful, since many people have poor eyesight and/or lack glasses, but i think for me, at least for all my use cases, and i think i would suggest generally(??) a single global text-options would be both clearer for the user as to what is happening, and easier+quicker to use since it would only need to be set in one place, and would remove the need for copy-settings-to (for instance, which would help the user aswell)

** middle + bottom share new-window, and minimize
The maximize option also has, i think, feature overlap with workspaces

Again im not aquainted with your users, but i would suggest simplification is possible. One possibility, would be to remove things from the middle window menus, leaving them with 'Sync / Move to / Close.' Placing 'Move to / Close' together also places in the user mind as being related. This simplification to 'Sync / Move to / Close' assumes text-options were a single global option -- not that they need to be at all, just that there are several places where i think the interface can be streamlined, and these are suggestions, i hope, moving towards that

I realize that you've already made large changes with version 4, and while these initial ideas are easy to suggest, they are not even fully formed, and by no means easy to do

---

Please note: these ui suggestions don't cause me any day-to-day problems, but issue #1879, and the crashing is causing me difficulties, and the text copying suggestion would make me happy :+1: 

Thanks so much for all you've done

**Comment by tse-gratis** [comment]
Would just like to say And Bible is fantastic software, and helps me a lot! Particularly the multi window feature, and language selection

I do advertise And Bible by word of mouth, and have given it to others to use

Thanks so much, with much appreciation to all you're doing!

**Comment by tuomas2** [comment]
Now this is very long issue. Can you summarize? 

Or perhaps @agrogers  or someone else can make me a clear short ticket out of this :-)

**Comment by agrogers** [comment]
Hi. You can see above the separate issues I have raised from this ticket. It is always good practice to open multiple small issues than one big issue. That's why its taken me a while to respond. Some other comments below that did not need a separate issue.

Thanks for your feedback. We do appreciate it.

> Since the text selection interface has the 'Bookmark / Compare' options, and this pops up when the user does press-and-hold on text, i think it would be great for the user, user interface consistency, and the program code if this was - THE ONLY - way to do bookmarks and verse comparison. Having tap-hold + single tap bookmarks + double tap for full screen + ... makes interaction harder.   #1880

I agree that more ways of doing things is not always helpful. These options are slightly different.
1. The single click is what I believe the **new** user expects. Something should happen when we click a verse. Now the most common things are accessed very easily with just one click. 
2. The long-click and select options have several problems. First, they take longer to execute. Second, a new user is less likely to find the option. Third, it doesnt show a dialog but a very limited menu. This method is only really useful for selecting text... which is why it has remained and allows us to bookmark parts of verses. Most users just want to bookmark a whole verse - so the single click is perfect for that.

> top + middle share text-options. Adding confusion is how the top right hamburger text-options only apply to last selected window, whereas middle text-options apply to a specific windows. To note i do actually find text sizes useful, since many people have poor eyesight and/or lack glasses, but i think for me, at least for all my use cases, and i think i would suggest generally(??) a single global text-options would be both clearer for the user as to what is happening, and easier+quicker to use since it would only need to be set in one place, and would remove the need for copy-settings-to (for instance, which would help the user aswell)

I think you might have misunderstood how these settings work. I agree that they look like they do the same thing. And i agree that sometimes they do do the same thing. But they are different. Check out this video - it might help explain it: https://www.youtube.com/watch?v=N7awlmZXFdQ

> ** middle + bottom share new-window, and minimize. The maximize option also has, i think, feature overlap with workspaces

Yes, this is duplicate functionality but it is there for a reason. It is possible to turn of the window buttons (the button that appears in the top right of each window). When that is turned off the only way to edit a window setting is through the window button bar at the bottom of the screen.

I have given some thought to trying to simplify these options and reduce the number of popup menus popping up other menus. Here was one attempt https://github.com/AndBible/and-bible/issues/1199 
The challenge is there are a lot of options - some you want to access regularly and others your rarely access. It's tricky to simplify - but thanks for your suggestions.

**Comment by tuomas2** [comment]
Maybe can close this now and continue discussion on new created tickets?

**Comment by agrogers** [comment]
Let's give it a day or two first.

___________________________
*www.TenOutOfTen.org* <https://www.TenOutOfTen.org>
***@***.***
(+61) 0477 174 193 (Australia)
(+95) 09 250018669 (Myanmar)

**Comment by tse-gratis** [comment]
These are excellent replies

I think the root problem is i wanted a different bible app. The video says:

> "The only study bible you need"
> "These settings are nearly the same, so it is easy to get them confused"

At the root level, study bibles are complex, and even paper study bibles have confusing things, that take time. So the video says "it is easy to get them confused." But even the best possible bible study app will not be able to fully escape this

Study bibles are harder to use and harder to make

---

The people we most often give bible apps cannot read in their own language. So i choose the faith comes by hearing bible apps for them. They are simple and the audio quality is good

For myself, and some others, i choose And Bible, because they are reading the bible. I think the issue is reading is different to studying

---

For example:

This is an excellent point:
> The single click is what I believe the new user expects. Something should happen when we click a verse. Now the most common things are accessed very easily with just one click

But 99.9% of my usecase is reading, so when i accidentally create bookmarks, i delete them, because they conflict with my usecase

But for users who want a study app, i agree, single-click dialog box can be the correct decision

**Comment by agrogers** [comment]
Hi. I very rarely  get the verse action dialog to popup when i dont want it
(full screen is an exception) and  i dont think i have ever accidentally
created a bookmark. So i am not quite following your work flow...

___________________________
*www.TenOutOfTen.org* <https://www.TenOutOfTen.org>
***@***.***
(+61) 0477 174 193 (Australia)
(+95) 09 250018669 (Myanmar)

**Comment by tse-gratis** [comment]
Thanks,

To stay on the main point, i agree when you said:

> Now the most common things are accessed very easily with just one click

If the user frequently creates bookmarks, then single click makes it easy, if the user doesn't, such as myself, then UI decisions are different

So you rightly corrected me previously. I wanted to make UI suggestions, some of them are independent of usecase, but most depend on what And Bible is optimizing for. I would need to ask where is And Bible aiming to place itself -- Logos Bible software, or a no-distraction ereader?

The Mac OS of bible software, or the Microsoft Word? Clear and pared back, or fully featured? Extra features have user interface costs

Single-click is just an example

---

To carry on with the single-click verse action dialog, these guys made a similar choice. Not on single-click, but double click and text select:
https://www.biblegateway.com/passage/?search=Matthew%203

I don't like it. If I'm interacting with text, say via selection, then the text is important to me, and this dialog hides the adjacent context

If I want to carry on reading, this interface means i probably need to cancel the text selection just made...

I would say for almost all users of this website, that was the wrong choice, even the dialog is not single-click. But there is going to be a set of users that use this. For those users the And Bible verse action dialog is already good and i would only suggest a minor change

**Comment by tse-gratis** [comment]
I think it would be good to discuss where And Bible is heading with its interface as a whole. Aiming for Logos Bible software might be right, but the extra complexity of that interface means a lot more work to try and reduce complexity for the user

**Comment by tse-gratis** [comment]
This is the reason:

> I’ve A/B tested [reduced functionality] at scale with millions of users across different verticals globally. In many situations, focusing on core functions and clear information presentation helps guide the user to using the site as primarily intended based on their primary intent. It’s tough for most product teams to realize that most users really don’t want all that additional functionality they built

**Comment by agrogers** [comment]
I believe that. Most people only use a fraction of the features in
Microsoft Word for example. Microsoft want people to buy their product so
they need to keep this in mind.

For us though no one is buying the app and we get no remuneration at all.
We just want an app that does what we need. It's great If other people find
it helpful. And this definitely factors into decision making processes. But
ultimately if we don't like it we won't give our time to work on it.

So the reference you sited is not completely relevant to us.

**Comment by tse-gratis** [comment]
I suppose like the difference between a power user and a 'normal' user?

What you did in #1199 is the gold standard of UI evaluation -- a user case study, and that same approach of standing back to see what they do

There are principles that help make a UI work well for both sets of users, but maybe user case studies would be a better way to go

Let me keep quiet, since my suggestions are in someways the antithesis of #1735 -- I don't think per window line-spacing and hyphenation options are what most users want. I maybe wrong. Though even complex interfaces such as blender don't provide this. Don't get me wrong, I could be happy to provide it, though I would do it a different way
